source/extensions/grpc_credentials/file_based_metadata/config.cc

```diff
@@ -69,11 +69,10 @@ FileBasedMetadataAuthenticator::GetMetadata(grpc::string_ref, grpc::string_ref,
   if (!config_.header_key().empty()) {
     header_key = config_.header_key();
   }
-  TRY_ASSERT_MAIN_THREAD {
+  TRY_NEEDS_AUDIT {
```

Comment:
@chaoqin-li1123 note that this had to be reverted to TRY_NEEDS_AUDIT as I think we are calling an exception-throwing function in something other than the main thread.

Reply:
Should we create an issue to track this then? Not sure if we are routinely auditing these sites